microsoft-edge/webview2/concepts/windowed-vs-visual-hosting.md

```diff
@@ -1,436 +1,185 @@
-* [ICoreWebView2CompositionController::CursorChanged event (add](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller#add_cursorchanged), [remove)](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller#remove_cursorchanged)
-* [ICoreWebView2CompositionController::SystemCursorId property (get)](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller#get_systemcursorid)<!--no put-->
-* [ICoreWebView2CompositionController::SendMouseInput method](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller#sendmouseinput)
-  * [COREWEBVIEW2_MOUSE_EVENT_KIND enum](/microsoft-edge/webview2/reference/win32/icorewebview2#corewebview2_mouse_event_kind)
-  * [COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS enum](/microsoft-edge/webview2/reference/win32/icorewebview2#corewebview2_mouse_event_virtual_keys)
-* [ICoreWebView2CompositionController::SendPointerInput method](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller#sendpointerinput)
-  * [COREWEBVIEW2_POINTER_EVENT_KIND enum](/microsoft-edge/webview2/reference/win32/icorewebview2#corewebview2_pointer_event_kind)
-* [ICoreWebView2Environment::CreateCoreWebView2PointerInfo method](/microsoft-edge/webview2/reference/win32/icorewebview2environment3#createcorewebview2pointerinfo)
-  * [ICoreWebView2PointerInfo interface](/microsoft-edge/webview2/reference/win32/icorewebview2pointerinfo)
+#### Advantages and disadvantages
 
----
+Visual hosting allows for (and requires) more granular control of layout.  When using this approach, the app needs specific handling of window management and rendering APIs.
+
+For example, when the user resizes the window, you must define how the WebView2 control scales in relation to the whole webpage, such as making the WebView2 scale twice as much as the app.
+<!-- todo:
+why would resizing require scaling the WebView2?
+what does "whole webpage" mean here?
+if scaling is required, why is the WebView2 scaling twice as much?
+-->
 
 
 <!-- ------------------------------ -->
-#### Accessibility
+#### APIs for Visual hosting
 
-By default, WebView2 will show up in the accessibility tree as a child of the parent `HWND`.  WebView2 provides an API to better position the WebView2 content relative to other elements in the app.
+For a list of APIs that can be used when configuring WebView2 in a Visual hosting environment, see [Rendering WebView2 using Composition](./overview-features-apis.md#rendering-webview2-using-composition) in _Overview of WebView2 features and APIs_.
 
-##### [.NET/C#](#tab/dotnetcsharp)
 
-Not applicable.
+<!-- ====================================================================== -->
+## Compatibility and constraints
 
-##### [WinRT/C#](#tab/winrtcsharp)
+Key compatibility limitations include the operating system and rendering in framework and non-framework apps.<!-- todo: what are the constraints about rendering, in framework and non-framework apps? -->
```

Comment:
what are the constraints about rendering, in framework and non-framework apps?

Reply:
Closing this "Outdated" comment.  
[Fresh Comment](https://github.com/MicrosoftDocs/edge-developer/pull/3271/files#r1777399077)